src/solvespace.h

```diff
@@ -172,16 +188,37 @@ enum class SolveResult : uint32_t {
     TOO_MANY_UNKNOWNS        = 20
 };
 
+#ifdef LIBRARY
+#   define ENTITY EntityBase
+#else
+#   define ENTITY Entity
+#endif
 
+typedef IdList<ENTITY,hEntity> EntityList;
 #include "sketch.h"
+
+#ifdef LIBRARY
+#   define CONSTRAINT ConstraintBase
+#   define GROUP GroupBase
+#else
+#   define CONSTRAINT Constraint
+#   define GROUP Group
+#endif
+
+#ifndef LIBRARY
 #include "ui.h"
+#endif
 #include "expr.h"
 
-
 // Utility functions that are provided in the platform-independent code.
-class utf8_iterator : std::iterator<std::forward_iterator_tag, char32_t> {
```

Comment:
I got a deprecated warning, and fixed it but I didn't actually test this `utf8_iterator` code, so, note to self: revert or test this